File: algorithms/Codility-master/Codility-master/7_StacksQueues/StoneWall.py
# 100%
def solution(H):
    # write your code in Python 3.6
    if len(H) < 2: return len(H)
    stack = []
    count = 0
    
    for item in H:
        
        while len(stack) != 0 and stack[-1] > item:
            stack.pop() 
        
        if len(stack) != 0 and stack[-1] == item:
            continue
        
        stack.append(item)
        #print(stack)
        count += 1
                    
    return count
                    
            

File: algorithms/Codility-master/Codility-master/7_StacksQueues/StoneWall.js

// 100%
function solution(H) {
    // write your code in JavaScript (Node.js 8.9.4)
    if (H.length < 2 )return H.length;
    let stack = [];
    let count = 0;
    for (let i=0; i<H.length; i++){
        while((stack.length != 0) && (stack[stack.length-1]>H[i])){
            stack.pop();
        }
        if ((stack.length !=0) && (stack[stack.length-1] == H[i])) continue;
        stack.push(H[i]);
        count += 1
    }
    return count;
}